Apps/Database/src/Delegates/IMessagingVerificationDelegate.cs

```diff
@@ -29,8 +29,9 @@ public interface IMessagingVerificationDelegate
         /// Inserts a Message Verification and any associated records.
         /// </summary>
         /// <param name="messageVerification">The message verification to insert.</param>
+        /// <param name="commit">If set to true the database changes will be persisted immediately.</param>
         /// <returns>Returns the guid of the saved message verification.</returns>
-        Guid Insert(MessagingVerification messageVerification);
+        Guid Insert(MessagingVerification messageVerification, bool commit = true);
```

Comment:
I'd very much that every time we touch an interface like that, we make it async and pass a cancellation token; unless it is has too much aftershocks, we can evaluate that it may not worth the time.

Reply:
As mentioned about the `.GetAwaiter` fix above, there is quite a bit of code and tests that need to change to accommodate the shift to async I'm happy to do it but I this ticket's time already blew out of proportion.